dask_jobqueue/remote_slurm.py

```diff
@@ -0,0 +1,174 @@
+        return out["job_id"]
+
+    async def close(self):
+        logger.debug("Stopping worker: %s job: %s", self.name, self.job_id)
+        # https://slurm.schedmd.com/rest_api.html#slurmctldCancelJob
+        try:
+            response = await self.api_client_session.delete(
+                f"{self.api_url}slurm/v0.0.36/job/{self.job_id}"
+            )
+            response.raise_for_status()
+        except aiohttp.ClientError as e:
+            logger.exception("SLURMJob request failed.")
+            raise RuntimeError from e
+        else:
+            return await response.json()
+
+    @classmethod
+    def _close_job(cls, job_id):
+        # Overriding to please weakref in Job.start, not used.
+        pass
+
+
+class RemoteSLURMCluster(SLURMCluster):
+    __doc__ = """ Launch Dask on a SLURM cluster
+
+    Parameters
+    ----------
+    api_url: str
+        A url (ending in /) pointing to the SLURM REST API
+        See https://slurm.schedmd.com/rest_api.html
+    queue : str
+        Destination queue for each worker job. Passed to `#SBATCH -p` option.
+    project : str
+        Accounting string associated with each worker job. Passed to `#SBATCH -A` option.
+    {job}
+    {cluster}
+    walltime : str
+        Walltime for each worker job.
+    job_cpu : int
+        Number of cpu to book in SLURM, if None, defaults to worker `threads * processes`
+    job_mem : str
+        Amount of memory to request in SLURM. If None, defaults to worker
+        processes * memory
+    job_extra : list
+        List of other Slurm options, for example -j oe. Each option will be prepended with the #SBATCH prefix.
+
+    Examples
+    --------
+    >>> from dask_jobqueue import SLURMCluster
+    >>> cluster = RemoteSLURMCluster(
+    ...     api_socket_path='/var/run/slurmrestd.socket.',
+    ...     queue='regular',
+    ...     project="myproj",
+    ...     cores=24,
+    ...     memory="500 GB"
+    ... )
+    >>> cluster.scale(jobs=10)  # ask for 10 jobs
+
+    >>> from dask.distributed import Client
+    >>> client = Client(cluster)
+
+    This also works with adaptive clusters.  This automatically launches and kill workers based on load.
+
+    >>> cluster.adapt(maximum_jobs=20)
+    """.format(
+        job=job_parameters, cluster=cluster_parameters
+    )
+
+    def __init__(
+        self,
+        api_client_session: aiohttp.ClientSession,
+        *args,
+        **kwargs,
+    ):
+        # Set into kwargs, they'll be passed through `**job_kwargs` into RemoteSLURMJob
+        kwargs["api_client_session"] = api_client_session
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def with_socket_api_client(
+        cls, socket_api_path: str, *args, **kwargs
+    ) -> "RemoteSLURMCluster":
+        kwargs["api_url"] = "http://localhost/"
+        return cls(
+            api_client_session=aiohttp.ClientSession(
+                connector=aiohttp.UnixConnector(path=socket_api_path)
```

Comment:
This session never gets closed. This is the warning you are seeing.

Reply:
Passing the kwargs instead of the ClientSession did the trick. Also reformatted the calls to be a bit less verbose with the try...except. :)